py/picca/io.py

```diff
@@ -545,21 +548,39 @@ def read_from_desi(nside,in_dir,thid,ra,dec,zqso,plate,mjd,fid,order):
             iv = iv.sum(axis=0)
             w = iv>0
             fl[w]/=iv[w]
-            d = forest(b_ll,fl,iv,t,ra[wt][0],de[wt][0],ztable[t],exp[wt][0],night[wt][0],fib[wt][0],order)
+            reso_sum = b_reso[wt].sum(axis=0)
+            reso= sp.clip(reso_sum,1.0e-6,1.0e6)
+            rms_in_pixel = (sp.sqrt(1.0/2.0/sp.log(reso[len(reso)/2][:]/reso[len(reso)/2-1][:]))
+                            + sp.sqrt(4.0/2.0/sp.log(reso[len(reso)/2][:]/reso[len(reso)/2-2][:])))/2.0
+            reso_in_km_per_s=spectral_resolution(rms_in_pixel,b_ll)
+            diff = b_ll*0.0
+            d  = forest(b_ll,fl,iv,t,ra[wt][0],de[wt][0],ztable[t],exp[wt][0],night[wt][0],fib[wt][0],order,diff,reso_in_km_per_s)
             ### R
             iv = r_iv[wt]
             fl = (iv*r_fl[wt]).sum(axis=0)
             iv = iv.sum(axis=0)
             w = iv>0
             fl[w]/=iv[w]
-            d += forest(r_ll,fl,iv,t,ra[wt][0],de[wt][0],ztable[t],exp[wt][0],night[wt][0],fib[wt][0],order)
-            ### Z
+            reso_sum = r_reso[wt].sum(axis=0)
```

Comment:
Is this doing what you want? Shouldn't you first reconstruct the 2D resolution matrix?

Reply:
So far the 2D resolution matrix does not really exist in the simulations. What we have is a gaussian PSF. But you are right I'm going to change the code I will introduce a function 
spectral_resolution_desi( reso_sum,b_ll) 
This will be allow us to change the code once a "true" 2D resolution matrix will be available and anayway the code will be more elegant for sure!